src/codegen/compile.ml

```diff
@@ -2926,6 +2941,21 @@ module Blob = struct
 
   let dyn_alloc_scratch env = alloc env ^^ payload_ptr_unskewed
 
+  let clear env =
+    Func.share_code1 env "blob_clear" ("x", I32Type) [] (fun env get_x ->
+      let (set_ptr, get_ptr) = new_local env "ptr" in
+      let (set_len, get_len) = new_local env "len" in
+      get_x ^^
+      as_ptr_len env ^^
+      set_len ^^
+      set_ptr ^^
+      get_len ^^
+      from_0_to_n env (fun get_i ->
+        get_ptr ^^ get_i ^^ G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
+        compile_unboxed_const 0l ^^
+        G.i (Store {ty = I32Type; align = 0; offset = 0l; sz = Some Wasm.Types.Pack8})))
```

Comment:
Would it be hard to optimise this to write i64 zeroes, plus a couple of odd extra bytes in the end if necessary?

Even better, can't we use memory.fill? That's standard now, and I think Wasmtime supports it.

Reply:
Yeah, I was actually planning to use i32 but either of those would better.

Reply:
The replica's appear to support MemoryFill, but our wasm lib is to old. We'll need to upgrade that first. I'll leave that to a separate PR.